src/pivot.coffee

```diff
@@ -584,6 +590,40 @@ callWithJQuery ($) ->
     ###
 
     pivotTableRenderer = (pivotData, opts, rendererType) ->
+        #If we've been asked to generate the totals from a meta-aggregator, walk the data tree here and do that
+        if pivotData.opts.totalsMetaAggregator
```

Comment:
Shouldn't this live in the PivotData class instead of the renderer? It's not a good division of responsibilities to have the renderer do data manipulation.

To make this work for both client and server mode, add this code as a method to PivotData, and call it from PivotData's constructor (to handle client mode), and call it in $.pivot as well (to handle server mode). In server mode, this method would be called twice, once when instantiating PivotData from PivotTable.js:renderTable(), and again from $.pivot, but the first time is essentially a no-op because the tree is empty.